internal/webhook/external/github.go

```diff
@@ -114,53 +114,57 @@ func githubHandler(
 			return
 		}
 
-		pe, ok := e.(*gh.PushEvent)
-		if !ok {
-			xhttp.WriteErrorJSON(w,
-				xhttp.Error(
-					fmt.Errorf("only push events are supported"),
-					http.StatusBadRequest,
-				),
+		switch e := e.(type) {
+		case *gh.PingEvent:
+			repo := e.GetRepo().GetHTMLURL()
+			logger.Debug("received ping event", "repo", repo)
+			xhttp.WriteResponseJSON(w,
+				http.StatusOK,
+				map[string]string{
+					"msg": fmt.Sprintf(
+						"ping event received, webhook is configured correctly for %s",
+						repo,
+					),
+				},
 			)
-			return
-		}
+		case *gh.PushEvent:
+			repo := e.GetRepo().GetHTMLURL()
+			logger.Debug("source repository retrieved", "name", repo)
```

Comment:
Being careful to not reduce the signal:noise ratio too much, I don't think this message is particularly valuable. I'd cut this.

Reply:
Also related to signal:noise ratio: I'm on the fence about this, but I think many (not all) of the things being logged as errors throughout the handler would be more appropriate as debug-level messages.

Error messages are useful when something is actually wrong with our software, but in cases where an expected header is missing, signature couldn't be verified, etc., our software is working correctly and the information is really only useful to someone who is actively troubleshooting.

cc @hiddeco for any opinion he may have on this.